imageregistry/v1/types.go

```diff
@@ -0,0 +1,346 @@
+	// accessing S3 storage.  This content takes precedence over content the operator
+	// automatically pulls from other locations, and it is merged into ImageRegistryPrivateConfiguration
+	ImageRegistryPrivateConfigurationUser = ImageRegistryPrivateConfiguration + "-user"
+
+	// ImageRegistryOperatorNamespace is the namespace containing the registry operator
+	// and the registry itself
+	ImageRegistryOperatorNamespace = "openshift-image-registry"
+
+	// ImageRegistryClusterOperatorResourceName is the name of the clusteroperator resource
+	// that reflects the registry operator status.
+	ImageRegistryClusterOperatorResourceName = "image-registry"
+
+	// Status Conditions
+
+	// OperatorStatusTypeRemoved denotes that the image-registry instance has been
+	// removed
+	OperatorStatusTypeRemoved = "Removed"
+
+	// StorageExists denotes whether or not the registry storage medium exists
+	StorageExists = "StorageExists"
+
+	// StorageTagged denotes whether or not the registry storage medium
+	// that we created was tagged correctly
+	StorageTagged = "StorageTagged"
+
+	// StorageLabeled denotes whether or not the registry storage medium
+	// that we created was labeled correctly
+	StorageLabeled = "StorageLabeled"
+
+	// StorageEncrypted denotes whether or not the registry storage medium
+	// that we created has encryption enabled
+	StorageEncrypted = "StorageEncrypted"
+
+	// StoragePublicAccessBlocked denotes whether or not the registry storage medium
+	// that we created has had public access to itself and its objects blocked
+	StoragePublicAccessBlocked = "StoragePublicAccessBlocked"
+
+	// StorageIncompleteUploadCleanupEnabled denotes whether or not the registry storage
+	// medium is configured to automatically cleanup incomplete uploads
+	StorageIncompleteUploadCleanupEnabled = "StorageIncompleteUploadCleanupEnabled"
+
+	// VersionAnnotation reflects the version of the registry that this deployment
+	// is running.
+	VersionAnnotation = "release.openshift.io/version"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []Config `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Config is the configuration object for a registry instance managed by
+// the registry operator
+type Config struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec ImageRegistrySpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// +optional
+	Status ImageRegistryStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+}
+
+type ImageRegistrySpec struct {
+	// ManagementState indicates whether the registry instance represented
```

Comment:
i think @gabemontero learned that all our godoc should use lowercase for field names (so it matches what a user would put in the yaml/json)

Reply:
that is correct